### DISCUSSION TITLE
How to create a smooth loop?

### DISCUSSION THREAD

**Discussion by gilles4nat** [katspaugh/wavesurfer.js#3753] [Q&A]
**wavesurfer.js version :**  7.7.15

I've been trying for a while to loop a region without any glitch (when the playback returns to the beginning of the region).

I know that using the WebAudio backend works perfectly for this. However, with this backend, when you change the playback speed, the pitch is not preserved. And of course, in my project, I need to be able to create smooth loops **and** change the tempo without affecting the pitch. I have tried several solutions to preserve the pitch with WebAudio, but without success. So I am reverting to the default backend (MediaElement).

I read that there was a function playLoop(). But it seems to no longer exist in version 7.
Do you have any ideas on how to loop a region smoothly, with the default Backend (MediaElement)?

Currently, I am doing like this:

`wsRegions.on('region-in', (region) => {
            loopRegion = region;
        });`

`wsRegions.on('region-out', (region) => {
            wavesurfer.setTime(loopRegion.start);
        });`

**Comment by katspaugh** [comment]
You could try (with Web Audio) to connect the audio buffer source to a delay node and modulate the delay speed with a saw wave to achieve a pitch shift.

**Reply by gilles4nat** [reply]
Thanks! I'm not very comfortable with WebAudio yet, but I'll give it a try!
